### PR TITLE
Fix install.sh script

### PR DESCRIPTION
**Pull request by bnallapeta**
With the current script, was running into the below issue. Fixed it by replacing the postchecks function with a simpler check of deployments instead of pods. 

```
Checking deployment
  - api-ci-ln-gpg06s2-72292-origin-ci-int-gce-dev-rhcloud-com:
    - tekton-pipelines-controller: error: error executing jsonpath "{.items[0].metadata.name}": Error executing template: array index out of bounds: index 0, length 0. Printing more information for debugging the template:
	template was:
		{.items[0].metadata.name}
	object given to jsonpath engine was:
		map[string]interface {}{"apiVersion":"v1", "items":[]interface {}{}, "kind":"List", "metadata":map[string]interface {}{"resourceVersion":"", "selfLink":""}}

```

Tested it by running the script on a cluster-bot cluster.

```
[bnr@bnr pipelines-service]$ WORKSPACE_DIR=./gitops/sre ./images/cluster-setup/install.sh 
Extracting files under the kubeconfig dir and reading the content in each file 
  - mycluster.kubeconfig
    - api-ci-ln-r4sp9jb-72292-origin-ci-int-gce-dev-rhcloud-com --- default/api-ci-ln-r4sp9jb-72292-origin-ci-int-gce-dev-rhcloud-com:6443/kube:admin --- mycluster.kubeconfig 

Installing tektoncd components on the cluster via Openshift GitOps... 
  - api-ci-ln-r4sp9jb-72292-origin-ci-int-gce-dev-rhcloud-com:
application.argoproj.io/pipelines-service unchanged

Checking deployment
Checking the status of tekton-pipelines-controller: tekton-pipelines-controller exists!
tekton-pipelines-controller is Available and Ready! 
Checking the status of tekton-triggers-controller: tekton-triggers-controller exists!
tekton-triggers-controller is Available and Ready! 
Checking the status of tekton-triggers-core-interceptors: tekton-triggers-core-interceptors exists!
tekton-triggers-core-interceptors is Available and Ready! 
```

### CODE REVIEW

File: images/cluster-setup/install.sh [hunk 1]
```diff
@@ -122,33 +122,28 @@ install_tektoncd() {
...
   #checking if the pipelines and triggers pods are up and running
   for i in "${!clusters[@]}"; do
     switch_cluster
     printf "  - %s:\n" "${clusters[$i]}"
-    for pod in tekton-pipelines-controller tekton-triggers-controller tekton-triggers-core-interceptors; do
-      printf "    - %s: " "$pod"
-      wait_for_pod -n openshift-pipelines -l=app="$pod"
-      printf "OK\n"
+    for deploy in tekton-pipelines-controller tekton-triggers-controller tekton-triggers-core-interceptors; do
+      printf "Checking the status of %s: " "$deploy"
```
Comment (Roming22) [inline images/cluster-setup/install.sh:132]: `printf "....- %s:" "$deploy"`. Replace '.' with spaces, as spaces were not displayed properly.

File: images/cluster-setup/install.sh [hunk 2]
```diff
@@ -122,33 +122,28 @@ install_tektoncd() {
...
+
+      #a loop to check if the deployment exists
+      if ! timeout 300s bash -c "while ! kubectl get deployment/$deploy -n openshift-pipelines >/dev/null 2>/dev/null; do printf '.'; sleep 10; done"; then
```
Comment (Roming22) [inline images/cluster-setup/install.sh:135]: `printf "......- Deployed: "`. Replace '.' with spaces, as spaces were not displayed properly.

File: images/cluster-setup/install.sh [hunk 3]
```diff
@@ -122,33 +122,28 @@ install_tektoncd() {
...
+        printf "%s does not exist. Something is wrong.\n" "$deploy"
+      else
+        printf "%s exists!\n" "$deploy"
```
Comment (Roming22) [inline images/cluster-setup/install.sh:138]: `printf "OK\n"`

File: images/cluster-setup/install.sh [hunk 4]
```diff
@@ -122,33 +122,28 @@ install_tektoncd() {
...
+      fi
+
+      #a loop to check if the deployment is Available and Ready
+      if kubectl wait --for=condition=Available=true deployment/$deploy -n openshift-pipelines --timeout=100s >/dev/null 2>/dev/null; then
```
Comment (Roming22) [inline images/cluster-setup/install.sh:142]: `printf "......- Up and running: "`. Replace '.' with spaces, as spaces were not displayed properly.

File: images/cluster-setup/install.sh [hunk 5]
```diff
@@ -122,33 +122,28 @@ install_tektoncd() {
...
+        printf "%s is Available and Ready! \n" "$deploy"
```
Comment (Roming22) [inline images/cluster-setup/install.sh:143]: `printf "OK\n"`

File: images/cluster-setup/install.sh [hunk 6]
```diff
@@ -122,33 +122,28 @@ install_tektoncd() {
...
+      else
+        printf "%s is not Ready.\n" "$deploy"
```
Comment (Roming22) [inline images/cluster-setup/install.sh:145]: `printf "Not ready\n"`

File: images/cluster-setup/install.sh [hunk 7]
```diff
@@ -122,33 +122,28 @@ install_tektoncd() {
...
   printf "Checking deployment\n"
   #checking if the pipelines and triggers pods are up and running
   for i in "${!clusters[@]}"; do
     switch_cluster
     printf "  - %s:\n" "${clusters[$i]}"
-    for pod in tekton-pipelines-controller tekton-triggers-controller tekton-triggers-core-interceptors; do
-      printf "    - %s: " "$pod"
-      wait_for_pod -n openshift-pipelines -l=app="$pod"
-      printf "OK\n"
+    for deploy in tekton-pipelines-controller tekton-triggers-controller tekton-triggers-core-interceptors; do
```
Comment (Roming22) [inline images/cluster-setup/install.sh:131]: If any (or all) those deployment failed, I do not see an error code being returned when the script exits. That could trigger silent failures.

Comment (bnallapeta) [reply images/cluster-setup/install.sh:131]: Fixed.

**Comment by bnallapeta** [comment]
@Roming22 I added the exit_error when there is a error. 
But as for the other comments, I am checking two things - a. If the pod exists b. If its Ready (1/1). If I print 'OK' for both these checks, it won't be meaningful.
I can change it to 'Exists' and 'Ready' if you prefer to have single words instead of a sentence that I currently have.

**Review by Roming22 — APPROVED** [review]

**Pull request merged**